src/Sarif/CodeGenHints.json

```diff
@@ -329,22 +329,13 @@
       }
     }
   ],
-  "Notification.TimeUtc": [
-    {
-      "kind": "AttributeHint",
-      "arguments": {
-        "namespaceName": "Newtonsoft.Json",
-        "typeName": "JsonConverter",
-        "arguments": [ "typeof(Microsoft.CodeAnalysis.Sarif.Readers.DateTimeConverter)" ]
-      }
-    }
-  ],
   "Notification.Level": [
     {
       "kind": "EnumHint",
       "arguments": {
-        "typeName": "NotificationLevel",
-        "description": "Values specifying the level of a notification.",
+        "typeName": "FailureLevel",
+        "description": "Values specifying the level of a notification or result.",
+        "zeroValueName": "None",
         "memberNames": [
           "Warning",
           "Note",
```

Comment:
>Note [](start = 11, length = 4)

Since this is now clearly a severity, I suggest we order the enum values by severity: `{ None = 0, Note, Warning, Error }`. I will do that in my branch.

Reply:
While I'm here, since again I think this now maps nicely to what people think of as "severity", did you consider calling the type `SeverityLevel` or even just `Severity`? Did you reject the idea because "`FailureLevel.None` ("no failure") reads better to you than "`Severity.None`" ("no severity"???)?

---
In reply to: [252888798](https://github.com/Microsoft/sarif-sdk/pull/1249#discussion_r252888798) [](ancestors = 252888798)

Reply:
I took the ordering change. I left it FailureLevel for now, strictly to help emphasize the direct relationship with ResultKind.Fail (which isn't ResultKind.HasSeverity)

---
In reply to: [252889280](https://github.com/Microsoft/sarif-sdk/pull/1249#discussion_r252889280) [](ancestors = 252889280,252888798)